pkg/alertmanager/statefulset.go

```diff
@@ -624,6 +627,24 @@ func makeStatefulSetSpec(logger *slog.Logger, a *monitoringv1.Alertmanager, conf
 		configReloaderVolumeMounts = append(configReloaderVolumeMounts, configMount...)
 	}
 
+	if !version.LT(semver.MustParse("0.24.0")) {
+		clusterTLSConfig, err := clustertlsconfig.New(clusterTLSConfigDir, clusterTLSConfigSecretName(a.Name), a.Spec.ClusterTLS)
```

Comment:
(nit) since we're also calling `clustertlsconfig.New()` in operator.go, should we call it once there and pass the struct as a new argument to makeStatefulSet()?

Reply:
We could do that, the same is true for Web config as well. If we are changing the logic here, I would also suggest that we change it for web config.
Apart from that, I feel like we can leave it (initialize twice) as is and keep the function signature for `makeStatefulSet` minimal. WDYT?

Reply:
right, we can address this later and consistently for both.